File: Leetcode problems/longest_common_prefix.cpp
string longestCommonPrefix(vector<string>& strs) {
        if(strs.size()==0)
            return "";
        
        string prefix = strs[0];
        
        for(int i=1;i<strs.size();i++)
        {
            while(strs[i].find(prefix)!=0)
            {
                prefix = prefix.substr(0,prefix.length()-1);
            }
        }
        return prefix;
    }

File: Leetcode problems/longest_common_prefix.py
# Write a function to find the longest common prefix string amongst an array of strings.
# If there is no common prefix, return an empty string "".


def longestCommonPrefix(strs):
        if not strs: return ''
        m, M, i = min(strs), max(strs), 0
        for i in range(min(len(m),len(M))):
            if m[i] != M[i]: 
                break
            else: 
                i += 1
        return m[:i]


if __name__ == "__main__":
    strs=["app","apple","appreature"]
    print(longestCommonPrefix(strs))
